addons/event_sale/tests/test_event_sale.py

```diff
@@ -376,7 +384,200 @@ def test_ticket_price_with_pricelist_and_tax(self):
             'event_id': event.id,
             'event_ticket_id': event_ticket.id,
         })
-        self.assertEqual(so.amount_total, 660.0, "Ticket is $1000 but the event product is on a pricelist 10 -> 6. So, $600 + a 10% tax.")
+        self.assertAlmostEqual(so.amount_total, 6.6, msg="Ticket is $1000 but the pricelist says fixed price $6 -> So, $6 + a 10% tax.")
+
+    def test_ticket_price_with_price_list_extensive(self):
```

Comment:
This could benefit from being extracted into another test class, because the many utility functions are distracting us from the flow of the test.
Also, maybe some instances of events and products(templates) can be created in `setUp` and modified as necessary for the different tests?

Reply:
Yes the tests have been reworked following this idea.